extensions/amp-story-interactive/0.1/amp-story-interactive-abstract.js

```diff
@@ -827,4 +828,32 @@ export class AmpStoryInteractive extends AMP.BaseElement {
       }
     });
   }
+
+  /**
+   * Reorders options data to account for scrambled or incomplete data.
+   *
+   * @private
+   */
+  orderData_() {
+    const numOptionElements = this.getOptionElements().length;
+    const orderedData = new Array(numOptionElements);
+    for (let i = 0; i < this.optionsData_.length; i++) {
+      const {index} = this.optionsData_[i];
+      if (index >= 0 && index < numOptionElements) {
+        orderedData[this.optionsData_[i].index] = this.optionsData_[i];
+      }
+    }
```

Comment:
This looks like it could be a forEach for simplicity:

```js
this.optionsData.forEach((option) => {
      const {index} = option;
      if (index >= 0 && index < numOptionElements) {
        orderedData[option.index] = option;
      }
})
```

FYI `forEach`s also allow you to retrieve the index, though it's not needed in this context: `foo.forEach((bar, i) => {console.log(bar, "is at index", i);})`

Reply:
Will do

Reply:
Also I noticed I forgot to replace `this.optionsData_[i].index` with `index` on line 843, so I'll change that too